pylint/testutils/_primer/comparator.py

```diff
@@ -0,0 +1,41 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+from __future__ import annotations
+
+import json
+from collections.abc import Generator
+from pathlib import Path
+
+from pylint.testutils._primer.primer_command import Messages, PackageMessages
+
+
+class Comparator:
```

Comment:
Could you rename this to `PylintComparator`? At some point I want to make an `AstroidComparator` 😄 

If you're up for it perhaps we could even create an abstract class?

Reply:
No, using the `primer` on `astroid/main` to catch crashes there before we merge them. That would require a different `Comparator` I think.

Reply:
Yeah, but I thinking of maybe running the `pylint` test suite on some version on `astroid` `main` or something like that. The overall goal is to remove the need for `pylint-tested` label and have all PRs get that automatically.

Reply:
Yeah sure I understood that :) It make sense to test astroid more in the astroid CI, because pylint's CI is already crowded so I think it's a really good idea too.